src/Microsoft.AspNet.Identity/RoleManager.cs

```diff
@@ -20,13 +21,17 @@ namespace Microsoft.AspNet.Identity
     public class RoleManager<TRole> : IDisposable where TRole : class
     {
         private bool _disposed;
-        private HttpContext _context;
+        private readonly HttpContext _context;
 
         /// <summary>
-        /// Constructor
+        ///     Constructor
         /// </summary>
         /// <param name="store">The IRoleStore commits changes via the UpdateAsync/CreateAsync methods</param>
-        /// <param name="roleValidator"></param>
+        /// <param name="roleValidators">IEnumerable of role validators</param>
+        /// <param name="keyNormalizer">user property normalizers</param>
+        /// <param name="errors">IdentityErrorDescribers</param>
```

Comment:
@HaoK  can provide more insights here. The last I remember the optional ones were initialized to null and instantiated in the constructor to have  a cleaner constructor and a default behavior. Looks like they have been made mandatory again. I thought this was need to have a default behavior if the users did not want to use DI